From 07de3dde4f1386934775a4983e25fbedafd451ba Mon Sep 17 00:00:00 2001
From: zain wang <wzz@rock-chips.com>
Date: Wed, 28 Mar 2018 18:29:38 +0800
Subject: [PATCH] Documentation: dt-bindings: fusb302: add power role
 compatible

Power role compatible can configure the typec power role,
and we can enable try.role function as well, that can
change the power role to our wanted if possible.

Change-Id: I85a3c4d0c5a54b37573f5fb24fe107d88f385021
Signed-off-by: zain wang <wzz@rock-chips.com>
---
 Documentation/devicetree/bindings/mfd/fusb302.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Documentation/devicetree/bindings/mfd/fusb302.txt b/Documentation/devicetree/bindings/mfd/fusb302.txt
index 239994208931..72c24fac041f 100644
--- a/Documentation/devicetree/bindings/mfd/fusb302.txt
+++ b/Documentation/devicetree/bindings/mfd/fusb302.txt
@@ -8,6 +8,12 @@ Required properties:
 - vbus-5v-gpios: enable/disable 5v vbus output
 - vbus-other-gpios: enable/disable other high-voltage vbus output
 - int-n-gpios: I2C int pin
+- fusb302,role: typec port power role
+		(ROLE_MODE_DRP/ROLE_MODE_DFP/ROLE_MODE_UFP)
+- fusb302,try_role: enable try.role function, it would try your wanted
+		    role when detect.
+		    Valid when fusb302,role == ROLE_MODE_DRP.
+		    (ROLE_MODE_DFP/ROLE_MODE_UFP)
 
 example:
 	fusb1: fusb30x@22 {
@@ -18,6 +24,8 @@ example:
 		vbus-5v-gpios = <&gpio1 4 GPIO_ACTIVE_LOW>;
 		vbus-other-gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
 		int-n-gpios = <&gpio1 24 GPIO_ACTIVE_HIGH>;
+		fusb302,role = "ROLE_MODE_DRP";
+		fusb302,try_role = "ROLE_MODE_DFP";
 		status = "okay";
 	};
 
-- 
2.35.3

